doc: nrf: Adding documentation support for IEEE 802.11 sniffer (#13490)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sniffer_rx_operation.rst

.. _ug_nrf70_developing_raw_ieee_80211_packet_reception_using_monitor_mode:

> Raw IEEE 802.11 packet Reception using monitor mode

Author: Hi @ioannisg ,

Yes, Monitor mode is the generally accepted sniffer mode naming convention. Monitor mode is the naming convention used when the device is working as a pure sniffer mode device.

> Raw IEEE 802.11 packet Reception using monitor mode

Author: okay

###################################################

.. contents::
   :local:
   :depth: 2

> The nRF70 Series device supports the reception of IEEE 802.11 packets in the application layer.

Reviewer: ```suggestion
The nRF70 Series device supports Monitor mode, that is, listening for and receiving all IEEE 802.11 traffic on a given channel, not restricted to a certain SSID. With Monitor mode raw IEEE 802.11 packets can be received directly at the user application layer.
```

Author: okay

> The nRF70 device has to be configured in monitor mode to let the nRF70 Series device to operate as an 802.11 wireless packet sniffer.

Reviewer: ```suggestion
The nRF70 device has to be configured in monitor mode in order to operate as an 802.11 wireless packet sniffer.
```

Author: okay

> When sending the received packet from the nRF70 Series device up the stack, the Wi-Fi driver will append a raw RX header to the received IEEE 802.11 packet and send it up to be received over a raw socket in the application layer.

Reviewer: ```suggestion
In Monitor mode, raw 802.11 packets received at the nRF70 Series device may be presented directly at the host application layer. When forwarding the received IEEE 802.11 packets to the user application, the Wi-Fi host driver will append a raw RX header to each received packet and deliver it up to the application layer using the raw socket API.
```

Author: okay

> The raw RX header will provide receive parameters for the received packet, such as data rate, receive signal strength, frequency of reception, and packet type (whether it is Legacy, HT, VHT, or HE).

Reviewer: ```suggestion
The raw RX header holds the receive parameters for the received packet, such as data rate, receive signal strength, frequency of reception, and packet type (whether it is Legacy, HT, VHT, or HE).
```

Author: okay

Raw IEEE 802.11 packet reception using Monitor mode
###################################################

.. contents::
   :local:
   :depth: 2

The nRF70 Series device supports Monitor mode, that is, listening for and receiving all IEEE 802.11 traffic on a given channel, not restricted to a certain SSID.
With Monitor mode raw IEEE 802.11 packets can be received directly at the user application layer.
The nRF70 device has to be configured in Monitor mode in order to operate as an 802.11 wireless packet sniffer.

In Monitor mode, raw 802.11 packets received at the nRF70 Series device may be presented directly at the host application layer.
When forwarding the received IEEE 802.11 packets to the user application, the Wi-Fi host driver will append a raw RX header to each received packet and deliver it up to the application layer using the raw socket API.
The raw RX header holds the receive parameters for the received packet, such as data rate, receive signal strength, frequency of reception, and packet type (whether it is Legacy, HT, VHT, or HE).

.. _ug_nrf70_developing_enabling_802.11_monitor_mode:

Enabling Monitor mode
*********************

> To use the monitor mode sniffer feature in your applications, you must enable the :kconfig:option:`CONFIG_NRF700X_RAW_DATA_RX` Kconfig option in the project configuration.

Author: let me reword this accordingly

.. _ug_nrf70_developing_monitor_mode_operation:

Monitor mode operation
**********************

The nRF70 Series device supports IEEE 802.11 packet reception when Monitor mode is enabled in the nRF Wi-Fi driver.
To enable the Monitor mode of operation, you can use the ``NET_REQUEST_WIFI_MODE`` network management runtime API.
This API can be used to disable Monitor mode in the driver when IEEE 802.11 packet reception is not required by the sniffer application.

> If the nRF70 device is configured to operate in Monitor mode, Station mode connection is not permitted and the nRF70 Series device operates in a pure sniffer mode.

Reviewer: It is also important to clarify whether the Monitor mode may be used at the same time with TX Injection Mode.
If so we need to mention that here. Ideally, we need to mention that in the Raw TX mode documentation as well (e.g. comment that in Non-Connected mode a user can enable both Monitor and TX Injection...)

> If the nRF70 device is configured to operate in Monitor mode, Station mode connection is not permitted and the nRF70 Series device operates in a pure sniffer mode.

Author: Monitor mode can be enabled with TX-Injection mode, but if Monitor mode is enabled, connection establishment is not possible. yes - you are correct, non-connected mode , user can enable Monitor and TX-Injection

> If the nRF70 device is configured to operate in Monitor mode, Station mode connection is not permitted and the nRF70 Series device operates in a pure sniffer mode.

Reviewer: Thanks! Yes, this should be explained in the guide, so user can know what combinations are possible.

> When Monitor mode is enabled, you need to configure the operating channel.

Reviewer: ```suggestion
The user needs to configure the operating Wi-Fi channel on which the nRF70 device will monitor and receive 802.11 packets.
```

> This channel will be used as the default channel on which the nRF70 device captures 802.11 packets.

Reviewer: I guess this sentence can be deleted, as a repetition. Also I am not 100% sure I understand the meaning of "default" here.

When the device operates in Monitor mode, all 802.11 packets received on the configured channel will be sent up the stack by the nRF70 Series device and the Wi-Fi driver.

> To set the desired channel for 802.11 packet reception, you can use the ``NET_REQUEST_WIFI_CHANNEL`` network management API.

Reviewer: ```suggestion
To set the desired channel for 802.11 packet reception, use the ``NET_REQUEST_WIFI_CHANNEL`` runtime NET-management API.
```

Reviewer: It is important to clarify, also
- whether the channel configure API can be used at any time, or whether there are restrictions (e.g. when Monitor mode is not yet enabled, etc.)
- whether there needs to be a certain sequence (ENABLE, SET_CHANNEL commands) or not

Author: will update this table. The set channel command is not restricted to the situation wherein Monitor mode is not enabled. However, I will run this comment by the UMAC team to confirm the operation

Author: @ioannisg - Not sure why i wrote that comment that way. I myself found it hard to fathom what I had written. 

Set channel has to be invoked when either TX-Injection mode (station non-connected) or Monitor mode is enabled. the channel configured will be the channel on which packets are sniffed for Monitor mode and is the channel on which RAW packets are transmitted when station is not connected. Let me think how to re-word this.

Reviewer: Thanks. IMHO it is valuable to clarify whether the channel can simply change in runtime, with the respective API call, independent of other configuration, or whether there needs to be a specific command sequence. This helps the users understand the steps they need to take.

See the :ref:`wifi_shell_sample` sample for more information on configuring mode and channel settings for raw packet reception through shell commands.

> The following table lists the shell commands and network management APIs used to switch between operational modes and set channels:

Reviewer: ```suggestion
The following table lists the shell commands and NET-management APIs that are used to enable Monitor and TX injection mode and select the desired Wi-Fi channel:
```

.. list-table:: Wi-Fi packet reception network management APIs
   :header-rows: 1

   * - Network management APIs
     - Wi-Fi shell command
     - Example usage
     - Description
   * - net_mgmt(NET_REQUEST_WIFI_MODE)
     - ``wifi mode -i<interface instance> <configuration>``
     - ``wifi mode -i1 -m``
     - Configure interface instance 1 to Monitor mode
   * - net_mgmt(NET_REQUEST_WIFI_MODE)
     - ``wifi mode -i<interface instance> <configuration>``
     - ``wifi mode -i1 -mt``
     - Configure interface instance 1 to Monitor mode and TX injection mode
   * - net_mgmt(NET_REQUEST_WIFI_CHANNEL)
     - ``wifi channel -i<interface instance> <channel number to set>``
     - ``wifi channel -i1 -c6``
     - Set the raw transmission channel to ``6`` for interface 1

.. _ug_nrf70_developing_monitor_mode_receive_packet_metadata:

Monitor mode receive packet metadata
************************************

<TODO> - need to determine if radiotap header format is to be used or proprietary header.

.. _ug_nrf70_developing_monitor_mode_receive_operation:

Monitor mode receive operation
******************************

> The IEEE 802.11 packet captured by the nRF70 Series device will be sent up to the Wi-Fi driver which will incorporate an additional radio information header to the received packet and send it to the upper layer application.

Reviewer: ```suggestion
An IEEE 802.11 packet captured by the nRF70 Series device in Monitor mode will be sent up to the Wi-Fi driver which will append a radio information header to the received packet and present the resulting set to the user application.
```

> The packet can be received in the upper layer application by opening a raw socket to the Wi-Fi driver.

Reviewer: ```suggestion
To receive the captured traffic in Monitor mode, the user application layer needs to open a raw socket to the Wi-Fi driver.
```

Reviewer: Are there any specific raw socket configuration required to receive the capture traffic in mOnitor mode? If so, it would be nice to explain that here.

Author: It is a regular raw socket with ETH_P_ALL and SOCK_RAW input parameters for the socket fd API. Nothing different. But let me see how i can elaborate further
